textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/neextractor/NamedEntityExtractorTest.java

```diff
@@ -0,0 +1,154 @@
+        neExtractor.open();
+
+        ITuple iTuple=neExtractor.getNextTuple();
+        ITuple targetTuple=NEExtractorTestConstants.getTest1ResultTuple();
+      //  Assert.assertTrue(TestUtils.equalTo(iTuple, targetTuple));
+        Assert.assertTrue(neExtractor.getNextTuple() == null);
+        neExtractor.close();
+    }
+
+    /**
+     * Scenario 2: Get next tuple with multiple return
+     * Text: Microsoft, Google and Facebook are organizations
+     */
+    @Test
+    public void getNextTupleTest2() throws Exception {
+        List<ITuple> data = NEExtractorTestConstants.getTest2Tuples();
+        IOperator sourceOperator = TestHelper(data.get(0).getSchema(),data);
+
+        NamedEntityExtractor neExtractor = new NamedEntityExtractor(sourceOperator);
+        neExtractor.open();
+
+        ITuple iTuple = neExtractor.getNextTuple();
+        ITuple targetTuple=NEExtractorTestConstants.getTest2ResultTuple();
+     //   Assert.assertTrue(TestUtils.equalTo(iTuple,targetTuple));
+        Assert.assertTrue(neExtractor.getNextTuple() == null);
+        neExtractor.close();
+    }
+
+    /**
+     * Scenario 3: Get next tuple with multiple return with different idendities
+     * Text: Microsoft, Google and Facebook are organizations and Donald Trump and Barack Obama are persons.
+     */
+    @Test
+    public void getNextTupleTest3() throws Exception {
+        List<ITuple> data = NEExtractorTestConstants.getTest3Tuples();
+        IOperator sourceOperator = TestHelper(data.get(0).getSchema(),data);
+
+        NamedEntityExtractor neExtractor = new NamedEntityExtractor(sourceOperator);
+        neExtractor.open();
+
+        ITuple iTuple = neExtractor.getNextTuple();
+        ITuple targetTuple=NEExtractorTestConstants.getTest3ResultTuple();
+    //    Assert.assertTrue(TestUtils.equalTo(iTuple,targetTuple));
+        Assert.assertTrue(neExtractor.getNextTuple() == null);
+        neExtractor.close();
+    }
+
+
+
+
+    /**
+     * Scenario 4: Get next tuple with multiple return with different idendities in two sentence(field).
+     * Text: Microsoft, Google and Facebook are organizations and Donald Trump and Barack Obama are persons.
+     */
+    @Test
+    public void getNextTupleTest4() throws Exception {
+        List<ITuple> data = NEExtractorTestConstants.getTest4Tuples();
+        IOperator sourceOperator= TestHelper(data.get(0).getSchema(),data);
+
+
+        NamedEntityExtractor neExtractor = new NamedEntityExtractor(sourceOperator);
+        neExtractor.open();
+
+        ITuple iTuple =neExtractor.getNextTuple();
+        //TODO: discuss the definition of return field.
+        ITuple targetTuple=NEExtractorTestConstants.getTest3ResultTuple();
+     //   Assert.assertTrue(TestUtils.equalTo(iTuple,targetTuple));
+        Assert.assertTrue(neExtractor.getNextTuple() == null);
+        neExtractor.close();
+    }
+
+
+
+    public IOperator TestHelper(Schema schema, List<ITuple> data) throws Exception {
```

Comment:
- This helper is not required, instead you can set these up (Refer to setUp() method in DictionaryMathcerTest.java). 

- You can write a method getQueryResults() to get the results returned from your neExtractor.getNextTuple() method. (Refer to the same file) and pass them along with the exactResults to TestUtils.containsAllResults() method to compare them. 

Hope this helps! :)

Reference: https://github.com/TextDB/textdb/blob/team1-refactor-dictionary-matcher/textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcherTest.java